### PR TITLE
IA-1715: Convert LinkDetails to function component

### PR DESCRIPTION
**Pull request by quang-le**
One of the last 3 class components to get rid of

Related JIRA tickets : IA-1715

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Doc
na

## Changes

- Add useGetLinkDetails hook
- change props so the whole `link`is passed from the parent
- thanks to that, make `linkDetailsSubComponent` function that can be passed to the Table (and remove TS error)
- Call `useValidateLinks` directly from `LinkDetails` i.o passing the mutation from the parent
- Delete old class component
- Delete obsolete fetch function
- Fix Cypress test

## How to test

If you don't have links:
- Go to datasources > versions > make a copy of current version
- Go to Org Units > Datasources > Algorithm runs
- Run any algorithm. Select 2 versions of your datasource as origin and destination

Then:
- Go to Links
- Click Search
- Click on the "details" IconButton (the eye):
    - All data should be displayed, including map/geometry
    - you should be able to validate/invalidate any row 

TL;DR: Nothing should have changed

## Print screen / video

https://github.com/BLSQ/iaso/assets/38907762/dff2e30d-a7a1-485a-a291-b0ef8d63f78c

## Notes

Things that the reviewers should know: known bugs that are out of the scope of the PR, other trade-offs that were made.
If the PR depends on a PR in bluesquare-components, or merges into another PR (i.o. main), it should also be mentioned here

### CODE REVIEW

File: hat/assets/js/apps/Iaso/utils/requests.js
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/utils/requests.js:1]: ❤️

**Review by beygorghor — APPROVED** [review]
Working nice
